### PR TITLE
feat(mutable): add support for mutable image builds

### LINKED ISSUE #32: Publish immutable artifact tags + canary version to Docker registries
We want to have the ability to have deis/charts reference a specific immutable image which will enable us better traceability on issues that come into various components.

What we've done with efforts in other repositories is publish a version:
- `git-[short sha]` that will never be overwritten
- `canary` that will move to reference the latest version available of a component

Examples of the easiest way to roll this change out can be seen in deis/workflow#487, deis/builder#246, and deis/workflow-manager#11

### PR DESCRIPTION
**Pull request by smothiki**
closes https://github.com/deis/dockerbuilder/issues/32

### CODE REVIEW

**Comment by smothiki** [comment]
ping @mboersma @arschles @sgoings

File: Makefile [hunk 1]
```diff
@@ -34,12 +33,9 @@ docker-compile:
 	go build -o ${BINDIR}/bin/boot -a -installsuffix cgo -ldflags ${LDFLAGS} boot.go
 
 # For cases where we're building from local
-docker-build:
+docker-build:
```
Comment (sgoings) [inline Makefile]: There's a lonely space in here. :cry:

File: Makefile [hunk 2]
```diff
@@ -9,7 +9,7 @@ export GO15VENDOREXPERIMENT=1
 
 # SemVer with build information is defined in the SemVer 2 spec, but Docker
 # doesn't allow +, so we use -.
```
Comment (mboersma) [inline Makefile]: Can you remove this comment too? It doesn't relate to anything obvious now.

File: Makefile [hunk 3]
```diff
@@ -21,10 +21,9 @@ BINDIR := ./rootfs
 DEV_REGISTRY ?= $(eval docker-machine ip deis):5000
 DEIS_REGISTY ?= ${DEV_REGISTRY}/
 IMAGE_PREFIX ?= deis
+include versioning.mk
 # Kubernetes-specific information for RC, Service, and Image.
```
Comment (mboersma) [inline Makefile]: This comment is also detached from its code now and should be removed.

File: Makefile [hunk 4]
```diff
@@ -36,10 +35,7 @@ docker-compile:
 # For cases where we're building from local
 docker-build:
 	docker build --rm -t ${IMAGE} rootfs
-
-# Push to a registry that Kubernetes can access.
-docker-push:
-	docker push ${IMAGE}
+	docker tag -f ${IMAGE} ${MUTABLE_IMAGE}
```
Comment (mboersma) [inline Makefile:33]: The `-f` flag is deprecated and produces a warning in current versions of Docker. Maybe we should remove it--AFAICT the behavior is the same either way.

Comment (smothiki) [inline Makefile:33]: I was just copying this from existing files. Let me know if you want me to remove -f tag in all other PRs that I have opened along with this one

Comment (mboersma) [inline Makefile:33]: Nah, let's just leave it for now. It doesn't cause problems as-is, but I can't be certain that removing it won't cause any.

**Pull request merged**